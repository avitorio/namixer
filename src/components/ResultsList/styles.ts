import styled, { css } from 'styled-components'
import { SearchResults } from 'templates/Home'

export const Results = styled.ul`
  margin-top: ${({ theme }) => theme.spacings.large};
  max-width: 96rem;
  width: 100%;

  li {
    list-style: none;
    margin-bottom: ${({ theme }) => theme.spacings.small};
  }
`

export type DomainProps = Pick<SearchResults, 'status'>

const domainModifiers = {
  available: () => css`
    opacity: 1;
  `,
  searching: () => css`
    opacity: 1;
  `,
  taken: () => css`
    span,
    strong {
      opacity: 0.5;
    }
  `
}

export const Domain = styled.a<DomainProps>`
  ${({ status }) => css`
    height: 68px;
    left: 242px;
    top: 574px;
    color: #000;
    background: #ffffff;
    border-radius: 8px;
    max-width: 96rem;
    width: 100%;
    display: flex;
    align-items: center;
    padding: 24px;
    ${!!status && domainModifiers[status]};
  `}
`
