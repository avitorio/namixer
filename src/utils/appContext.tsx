import {
  Dispatch,
  SetStateAction,
  useState,
  useContext,
  createContext,
  ReactNode
} from 'react'

export interface IProviderProps {
  children?: ReactNode
}

type AppContextState = { domain: string }

type AppContextValue = {
  state: AppContextState
  setState: Dispatch<SetStateAction<AppContextState>>
}

export const AppContext = createContext<AppContextValue | undefined>(undefined)

export const AppProvider = (props: IProviderProps) => {
  const [state, setState] = useState({ domain: 'YourDomain.com' })

  return (
    <AppContext.Provider value={{ state, setState }}>
      {props.children}
    </AppContext.Provider>
  )
}

export function useAppContext() {
  const ctxValue = useContext(AppContext)
  if (ctxValue === undefined)
    throw new Error(
      'Expected an AppProvider somewhere in the react tree to set context value'
    )
  return ctxValue // now has type AppContextValue
  // or even provide domain methods for better encapsulation
}
