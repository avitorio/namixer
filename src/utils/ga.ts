// https://developers.google.com/analytics/devguides/collection/gtagjs/pages
export const pageview = (url: URL): void => {
  window.gtag('config', `${process.env.NEXT_PUBLIC_GA_TRACKING}`, {
    page_path: url
  })
}

type EventParams = { [key: string]: string | number }

type GTagEvent = {
  event: string
  params: EventParams
}

// https://developers.google.com/analytics/devguides/collection/gtagjs/events
export const event = ({ event, params }: GTagEvent): void => {
  window.gtag('event', event, {
    ...params
  })
}
