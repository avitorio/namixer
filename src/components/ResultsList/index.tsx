import { Session } from 'next-auth'
import InfiniteScroll from 'react-infinite-scroll-component'
import { SearchResults } from 'templates/Home'
import * as S from './styles'

export type ResultsListProps = {
  results?: SearchResults[]
  onSubmit?: () => Promise<void>
  hasNextPage?: boolean
  session?: Session | null
}

const ResultsList = ({
  results = [],
  onSubmit = () => new Promise(() => null),
  hasNextPage = false,
  session = null
}: ResultsListProps) => (
  <S.Results>
    <InfiniteScroll
      dataLength={results.length}
      next={onSubmit}
      hasMore={hasNextPage}
      loader={<h4>Loading...</h4>}
    >
      {results.map((result) => {
        if (result.domain) {
          return (
            <li key={result.domain}>
              <S.Domain status={result.status}>
                <div>
                  <strong>{result.domain}</strong>
                  {result.status === 'taken' && <span>Taken</span>}
                  {result.status === 'searching' && <span>Searching</span>}
                  {result.status === 'available' && <span>Available</span>}
                </div>
              </S.Domain>
            </li>
          )
        }
      })}
    </InfiniteScroll>
    {results.length > 0 && !session?.user?.name && (
      <h1>Register for more results</h1>
    )}
  </S.Results>
)

export default ResultsList
