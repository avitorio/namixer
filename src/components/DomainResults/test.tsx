import { render } from 'utils/test-utils'

import ResultsList from '.'

describe('<ResultsList />', () => {
  it('should render the heading', () => {
    const { container } = render(<ResultsList />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
