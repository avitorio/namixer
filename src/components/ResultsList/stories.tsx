import { Story, Meta } from '@storybook/react/types-6-0'
import ResultsList from '.'

export default {
  title: 'ResultsList',
  component: ResultsList
} as Meta

export const Default: Story = () => <ResultsList />
