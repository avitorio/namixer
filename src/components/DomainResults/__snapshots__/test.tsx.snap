// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsList /> should render the heading 1`] = `
.c0 {
  margin-top: 4.0rem;
  max-width: 96rem;
  width: 100%;
}

.c0 li {
  list-style: none;
  margin-bottom: 2.4rem;
}

<ul
  class="c0"
>
  <div
    class="infinite-scroll-component__outerdiv"
  >
    <div
      class="infinite-scroll-component "
      style="height: auto; overflow: auto;"
    />
  </div>
</ul>
`
